app/jobs/transfer_basespace_files.rb

```diff
@@ -11,8 +11,21 @@ def self.perform(sample_id, basespace_dataset_id, basespace_access_token)
     sample = Sample.find(sample_id)
     sample.transfer_basespace_files(basespace_dataset_id, basespace_access_token)
 
-    # TODO: Revoke the access token, so that it can no longer be used.
-    # Need to do this after ALL samples that use the access token can be uploaded.
+    # If ALL samples relying on this access token are done transferring files, revoke the access token.
+    samples_remaining = Sample.where(basespace_access_token: basespace_access_token, status: Sample::STATUS_CREATED)
+
+    # If multiple samples finish file transfer at exactly the same time, this block of code may be run multiple times.
+    if samples_remaining.empty?
+      begin
+        BasespaceHelper.revoke_access_token(basespace_access_token)
+      rescue
```

Comment:
Same thing here, you should catch a specific error. Otherwise, things like "method doesn't exist" will silently succeed. And also, we should log the error. `rescue... nil ... end` is a dangerous construction.